File: app/src/main/java/ru/nikitazhelonkin/coinbalance/data/api/coin/TRXApiService.java
package ru.nikitazhelonkin.coinbalance.data.api.coin;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import ru.nikitazhelonkin.coinbalance.data.api.response.TRXAssetResponse;
import ru.nikitazhelonkin.coinbalance.data.api.response.TRXWalletResponse;

public interface TRXApiService {

    @GET("v1/accounts/{address}")
    Single<TRXWalletResponse> balance(@Path("address") String address);

    @GET("v1/assets/{id}")
    Single<TRXAssetResponse> getAsset(@Path("id") String id);
}

File: app/src/main/java/ru/nikitazhelonkin/coinbalance/data/api/coin/TRXApiService.kt
package ru.nikitazhelonkin.coinbalance.data.api.coin

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import ru.nikitazhelonkin.coinbalance.data.api.response.TRXAssetResponse
import ru.nikitazhelonkin.coinbalance.data.api.response.TRXWalletResponse

interface TRXApiService {

    @GET("v1/accounts/{address}")
    fun balance(@Path("address") address: String): Single<TRXWalletResponse>

    @GET("v1/assets/{id}")
    fun getAsset(@Path("id") id: String): Single<TRXAssetResponse>
}
